README.md

```diff
@@ -84,18 +84,33 @@ Once you have implemented the tools, fill out the report below.
 
 ### Insights you may have had while implementing and comparing the algorithms. 
 
-*Describe this here.*
+Implementation wise, C is a little difficult to get started with.
+
+If you are used to just calling something like str.replace("\n",""),
+that is unfortunately not going to fly in C.
+
+Got the hang of it though, and playing around with pointers isn't so scary anymore.
 
 ### Problems encountered if any. 
 
-*Describe this here.*
+We malloc a lot in our parser, as we do not know how many pairs of fasta heads and sequences our file will contain.
+
+The solution seems to be to malloc n, then n *2, then n *4 etc., however it seems quite tedious to implement (and a little slower depending on input).
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+We have implemented tests in test/tests.c, testing the functionality of the naive and linear implementations. 
+
+The naive was implemented without any consideration. The linear KMP algorithm was implemented in a TDD manner.
+
+Both implementations were end-to-end stress tested with randomly generated 20 mb fasta files and lots of reads with success.
```

Comment:
Be careful here. Random strings might not be the best way to test these algorithms. Random strings tend not to share a lot of sequences, so you never compare far in the naive algorithm. The naive algorithm will usually behave within a factor of two of the best-case scenario if you give it random strings (something you can work out by assuming letters are sampled randomly, which gives you a geometric distribution of mismatch).

Anyway, the details are not important. Just keep in mind that random data 1) might not look like real data (that is rarely random) and 2) might trigger worst-case behaviour with a very low probability, so you never see it in the experiments.

It is always worthwhile to think about what kind of data will trigger different behaviour in an algorithm and then test with that kind of data as well.